### PR TITLE
DROOLS-3727 - Initial Kogito work branch into master

### PR DESCRIPTION
**Pull request by tiagodolphine**
Merge  Kogito branch (DROOLS-3727) into master details on [this PR](https://github.com/kiegroup/kie-wb-common/pull/2923)

Related PR's
https://github.com/kiegroup/kie-wb-common/pull/2923
https://github.com/kiegroup/kie-wb-distributions/pull/977

### CODE REVIEW

**Comment by mareknovotny** [comment]
it would be nice to exclude these new management from 7.x stream and have it optionally in our build only definitely exclude it from productized profile in downstream repositories.

**Comment by manstis** [comment]
@mareknovotny The changes to isolate the _core_ editor for re-use in Business Central and kogito (VSCode etc plugins) necessitated the introduction of new modules. Following merge of this PR whether the editors are being built for Business Central of kogito still requires the new modules. It is therefore not possible to exclude management of the new modules for anything on `master` (7.x community releases).

File: kie-bom/pom.xml [hunk 1]
```diff
@@ -2583,6 +2650,74 @@
...
+      </dependency>
+      <dependency>
+        <groupId>org.kie.workbench</groupId>
+        <artifactId>kie-wb-common-dmn-webapp-common</artifactId>
+        <version>${version.org.kie}</version>
+        <classifier>sources</classifier>
+      </dependency>
+      <dependency>
+        <groupId>org.kie.workbench</groupId>
+        <artifactId>kie-wb-common-dmn-webapp-kogito-common</artifactId>
```
[the remark this answers is not in the document]
Comment (jomarko) [reply kie-bom/pom.xml]: @mareknovotny I haven't go so deeply, I wanted just a comment in pom. spliting regular DMN and kogito DMN artifacts

**Comment by manstis** [comment]
Somewhere I saw a comment from @romartin agreeing with @mareknovotny comments. Given time to reflect I tend to agree too.. it's not _right_ that Business Central seems to have dependencies on `kogito` modules.. some naming changes may be in order and dependencies reviewed... I have proposed that perhaps @tiagodolphine could work on addressing this (in an email to those directly involved with this work)..... time will tell....

**Comment by hasys** [comment]
Hi @tiagodolphine,

I agree with @manstis, @romartin, @mareknovotny, that Business Central shouldn't depend on Kogito.

**Comment by romartin** [comment]
@tiagodolphine @manstis @hasys @mareknovotny 

Ok thanks for answering guys. As it seems we all agree on refactoring the kogito "root" modules, so making the BC agnostic to this "kogito" term, I think it will really be much better doing this before the merge, otherwise the artifacts, although probably just snapshots, will be exposed to the world... 

So if you agree @tiagodolphine , plz let's see what we can do to fix this, if you want some quick call just lemme know, but should be not hard to do it, at worst case we can just refactor the module names even, and just decuple what's trivial from those. WDYT?

Thx!

**Comment by manstis** [comment]
@tiagodolphine @romartin During my re-work of https://github.com/kiegroup/kie-wb-common/pull/2923 following @jomarko review I've [removed](https://github.com/kiegroup/kie-wb-common/commit/0c3bd50235bd8b04e9f66e665d0b99c2dc6cecd1#diff-29a763f0139c764d108ac5c19d3f898cL103) `kogito-xxx` dependencies from `-project-client` that should invalidate the need for this PR (if BPMN can make similar changes)...

I think let's wait for https://github.com/kiegroup/kie-wb-common/pull/2923 to be reviewed and complete "for merge" and re-access this PR.

**Comment by jomarko** [comment]
I do not see any response to my question, could we please add a leading comment into pom.xml files that would title dmn-kogito artifacts? Sure, if we can not do it, please tell me.

**Comment by manstis** [comment]
@jomarko DMN KOGITO artifacts split as requested.

**Pull request merged**